### PR TITLE
chore(ci): only run semgrep, fossa & snyk on branches owned by openfga org

### PR DESCRIPTION
**Pull request by rhamzeh**
These checks will always fail for external contributors, this change only makes them run on branches coming in from within the org, we will have to rely on the weaker webhook integration for outside PRs (these checks will still run on merge to main).

## Review Checklist
- [ ] I have added documentation for new/changed functionality in this PR or in [openfga.dev](https://github.com/openfga/openfga.dev)
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: .github/workflows/snyk.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ on:
 jobs:
   security:
     runs-on: ubuntu-latest
+    if: (github.repository_owner == 'openfga' && (github.actor != 'dependabot[bot]' && github.actor != 'snyk-bot'))
```
Comment (jon-whit) [inline .github/workflows/snyk.yaml:12]: Should we run snyk and semgrep even on dependabot PRs? Why did we ignore these actor(s) in the first place?

File: .github/workflows/semgrep.yaml [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
       - main
+      - master
 jobs:
   semgrep:
     name: Scan
     runs-on: ubuntu-latest
     container:
       image: returntocorp/semgrep
-    if: (github.actor != 'dependabot[bot]' && github.actor != 'snyk-bot')
+    if: (github.repository_owner == 'openfga' && (github.actor != 'dependabot[bot]' && github.actor != 'snyk-bot'))
```
Comment (craigpastro) [inline .github/workflows/semgrep.yaml:14]: I think the extra parenthesis are not required here. Should we get rid of them?

File: .github/workflows/snyk.yaml (hunk 1 above)
Comment (rhamzeh) [reply .github/workflows/snyk.yaml:12]: At the moment the secrets are not set for dependabot except for FOSSA, we'd need to renable once the secrets are added

**Review by craigpastro — APPROVED** [review]
LGTM

**Pull request merged**